Add db_index=True to object_pk and is_removed fields

atodorov opened the pull request.

atodorov commented:
TextField with db_index will fail on MySQL/MariaDB b/c there's no fixed size specified. Works fine on Postgres.  In a different library that was solved by changing object_pk to CharField(max_length=64). See https://github.com/bartTC/django-attachments/pull/81

Any objections to do the same here ?

Note2: the reported PASS result means there's no testing with MySQL here so that also needs to be added. I can work on it if the CharField change is accepted.

atodorov commented:
> TextField with db_index will fail on MySQL/MariaDB b/c there's no fixed size specified. Works fine on Postgres. In a different library that was solved by changing object_pk to CharField(max_length=64). See [bartTC/django-attachments#81](https://github.com/bartTC/django-attachments/pull/81)
> 
> Any objections to do the same here ?

@claudep ^^^

claudep commented:
I think that transforming `TextField` to `CharField(max_length=64)` for `object_id` should be fine for a vast majority of use cases. We could even describe how to keep the TextField `object_id` by a custom Comment model if really needed.
@carltongibson, @felixxm, do you think this would be an acceptable backwards incompatibility? My hopes of getting auto-limited MySQL index sizes (in the spirit of https://github.com/django/django/pull/8848/files) are now very thin.

atodorov commented:
I've converted `object_pk` field to `CharField` and updated the history in case you decide this looks good enough and want to merge. Thanks for reviewing.

felixxm commented:
@claudep I think it's fine. Docs changes are missing, e.g. https://github.com/django/django-contrib-comments/blob/ab3e0e6afa0af0a478ce024f3f398134da7d778f/docs/models.txt#L29-L33

claudep commented on HISTORY.rst:
```diff
@@ -8,6 +8,9 @@ Unreleased
 * Added the ``delete_stale_comments`` management command.
+* Added db_index to ``object_pk`` and ``is_removed`` fields.
+* Altered ``object_pk`` field to ``CharField(max_length=64)`` for
+  compatibility with MySQL.
```
I would be a little more verbose here. My suggestion:
`Altered object_pk from TextField to CharField(max_length=64) so that the field can be indexed on MySQL, too. Warning: if you attach comments to objects whose primary key is serialized to more than 64 characters, you should provide a custom Comment model (more about that in the documentation) with an appropriate object_pk field.`

claudep commented on django_comments/abstracts.py:
```diff
@@ -23,7 +23,7 @@ class BaseCommentAbstractModel(models.Model):
                                      related_name="content_type_set_for_%(class)s",
                                      on_delete=models.CASCADE)
-    object_pk = models.TextField(_('object ID'))
+    object_pk = models.CharField(_('object ID'), db_index=True, max_length=64)
```
@felixxm, do you think it's fine to use `db_index` here, or would it be better to use a more modern Meta `Index`?

claudep commented on HISTORY.rst:
```diff
@@ -8,6 +8,14 @@ Unreleased
+  comments to objects whose primary key is serialized to more than 64
+  characters, you should provide a custom Comment model
+  (more about that in the documentation) with an appropriate
+  ``object_pk``field.
```
missing space before "field".

claudep commented:
@atodorov, thanks for your patient work on this. I think it's good to go.
